homeassistant/components/media_player/harman_kardon_avr.py

```diff
@@ -0,0 +1,137 @@
+"""
+Support for interface with an Harman/Kardon or JBL AVR.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/media_player.harman_kardon_avr/
+"""
+import logging
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.media_player import (
+    SUPPORT_TURN_OFF, SUPPORT_VOLUME_MUTE, SUPPORT_VOLUME_STEP,
+    PLATFORM_SCHEMA, SUPPORT_TURN_ON, SUPPORT_SELECT_SOURCE,
+    MediaPlayerDevice)
+from homeassistant.const import (
+    CONF_HOST, CONF_NAME, CONF_PORT, STATE_OFF, STATE_ON, STATE_UNKNOWN)
+
+REQUIREMENTS = ['hkavr==0.0.5']
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = 'Harman Kardon AVR'
+DEFAULT_PORT = 10025
+
+SUPPORT_HARMAN_KARDON_AVR = SUPPORT_VOLUME_STEP | SUPPORT_VOLUME_MUTE | \
+                            SUPPORT_TURN_OFF | SUPPORT_TURN_ON | \
+                            SUPPORT_SELECT_SOURCE
+
+SOURCES = ["Disc", "STB", "Cable Sat", "Media Server", "DVR", "Radio", "TV",
```

Comment:
I suggest we move remove this constant from home assistant and use the `DEFAULT_SOURCES` constant in the hkavr library. Import that in the entity constructor.

Let the library handle the sources.

Reply:
It even looks like we can use the `avr.sources` property.